sphinx/templates/apidoc/package.rst_t

```diff
@@ -35,14 +35,15 @@ Submodules
 ----------
 {% if separatemodules %}
 {{ toctree(submodules) }}
-{%- else %}
+{% endif %}
+{%- if not separatemodules %}
 {%- for submodule in submodules %}
 {% if show_headings %}
 {{- [submodule, "module"] | join(" ") | e | heading(2) }}
 {% endif %}
 {{ automodule(submodule, automodule_options) }}
 {% endfor %}
-{%- endif %}
+{%- endif -%}
```

Comment:
I think using stripping mark at both start and end causes template maintainable. I guess the change above (changing "else" to independent "if" statement) came from this change, right? To keep templates clean, it is better to use one of them consistently.

How about the change like this?
```
@@ -43,9 +46,10 @@ Submodules
 {{ automodule(submodule, automodule_options) }}
 {% endfor %}
 {%- endif %}
-{% endif %}
+{%- endif %}

 {%- if not modulefirst and not is_namespace %}
+
 Module contents
 ---------------
```

Notes:
* Inserts a blank line before section headers to add a space only if the block needed.
* Add stripping-mark to the start of "endif" to strip trailing return codes.

I guess some additional change would be needed for this change. But it will keep our templates clean (if my understanding correct). Could you check this please?